frontend/src/bundles/talent-onboarding/components/bsa-badges-step/bsa-badges-step.tsx

```diff
@@ -0,0 +1,165 @@
+import {
+    Controller,
+    type ControllerFieldState,
+    type ControllerRenderProps,
+    type UseFormStateReturn,
+} from 'react-hook-form';
+import { useSelector } from 'react-redux';
+
+import {
+    Badge,
+    Checkbox,
+    FormControl,
+    FormHelperText,
+    Typography,
+} from '~/bundles/common/components/components.js';
+import {
+    // useAppDispatch,
+    useAppForm,
+    useCallback,
+    // useEffect,
+} from '~/bundles/common/hooks/hooks.js';
+import { type RootReducer } from '~/framework/store/store.package.js';
+
+//import { useFormSubmit } from '../../context/context.js';
+import { getRandomBadgeColor } from '../../helpers/helpers.js';
+//import { actions } from '../../store/talent-onboarding.js';
+import { type BsaBadgesStepDto } from '../../types/types.js';
+import { BsaBadgesStepValidationSchema } from '../../validation-schemas/validation-schemas.js';
+import styles from './styles.module.scss';
+
+//TODO: uncomment when context for submitting will be added
+const BsaBadgesStep: React.FC = () => {
+    const savedPayload = useSelector(
+        (state: RootReducer) => state.talentOnBoarding.bsaBadgesStep,
+    );
+
+    const bsaBadges = useSelector(
+        //TODO: get all user badges here, for now mock data
+        (state: RootReducer) => state.lms.bsaBadges,
+    );
```

Comment:
I think you don't need two useSelector's here.

Maybe this approach will be better

```
const { savedPayload, bsaBadges } =
    useAppSelector((state) => ({
      savedPayload: state.talentOnBoarding.bsaBadgesStep,
      bsaBadges: state.lms.bsaBadges,
    }));
```